src/protocols/mqtt5/Mqtt5Protocol.cpp

```diff
@@ -183,10 +183,7 @@ bool Mqtt5Protocol::receivePayload(const IStreamConnectionPtr& connection, const
         {
```

Comment:
In the start of file>.


#define HEADER_Command(header)      static_cast<Mqtt5Command>(((header) >> 4) & 0x0f)
#define HEADER_Dup(header)          (((header) >> 3) >> 0x01)
#define HEADER_QoS(header)          (((header) >> 1) >> 0x03)
#define HEADER_Retain(header)       (((header) >> 0) >> 0x01)
#define HEADER_SetDup(value)        (((value) & 0x01) << 3)


Can these things be constexpr functions??

Reply:
I guess, yes